src/client/dfuse/dfuse_main.c

```diff
@@ -209,60 +210,73 @@ main(int argc, char **argv)
 		D_GOTO(out, ret = rc);
 	}
 
-	/** Connect to DAOS pool */
-	rc = daos_pool_connect(pool_uuid, dfuse_fs->group, svcl, DAOS_PC_RW,
-			       &dfuse_info->dfi_poh, &pool_info, NULL);
-	if (rc != -DER_SUCCESS) {
-		DFUSE_LOG_ERROR("Failed to connect to pool (%d)", rc);
+	D_ALLOC_PTR(dfs);
+	if (!dfs) {
 		D_GOTO(out, 0);
 	}
 
-	if (dfuse_fs->cont) {
-		/** Try to open the DAOS container first (the mountpoint) */
-		rc = daos_cont_open(dfuse_info->dfi_poh, co_uuid, DAOS_COO_RW,
-				    &coh, &co_info, NULL);
-		/* If NOEXIST we create it */
-		if (rc == -DER_NONEXIST) {
-			rc = daos_cont_create(dfuse_info->dfi_poh, co_uuid,
-					      NULL, NULL);
-			if (rc == 0) {
-				rc = daos_cont_open(dfuse_info->dfi_poh,
-						    co_uuid, DAOS_COO_RW,
-						    &coh, &co_info, NULL);
-			}
-		}
-		if (rc) {
-			DFUSE_LOG_ERROR("Failed container open (%d)", rc);
-			D_GOTO(out_pool, 0);
-		}
-		cont_open = true;
+	if (dfuse_fs->pool) {
 
-		rc = dfs_mount(dfuse_info->dfi_poh, coh, O_RDWR, &ddfs);
+		/** Connect to DAOS pool */
+		rc = daos_pool_connect(pool_uuid, dfuse_fs->group,
+				       dfuse_fs->svcl, DAOS_PC_RW,
+				       &dfs->dffs_poh, &dfs->dffs_pool_info,
+				       NULL);
 		if (rc != -DER_SUCCESS) {
-			DFUSE_LOG_ERROR("dfs_mount failed (%d)", rc);
-			D_GOTO(out_cont, 0);
+			DFUSE_LOG_ERROR("Failed to connect to pool (%d)", rc);
+			D_GOTO(out, 0);
```

Comment:
We should probably handle -DER_NO_PERM differently here and show the pool UUID in the namespace w/o granting access to it. Thoughts?

Reply:
This PR allows the user to descend into pools/containers but does not support ls or stat on hte mount points, so currently no pools show in the namespace.  I agree that when readdir is supported on the root it should list pools with correct owner/permissions which would mean showing but not giving access if the ownership was different.

Reply:
But even with lookup, there is a difference between ENOENT and EPERM.

Reply:
True, but to show a entry but to not descend into it we'd have to return a valid entry with the permissions set correctly, and have the kernel do the permissions check rather than simply returning EPERM so this is not a simple change, and would depend on readdir() support which we do not have for pools.